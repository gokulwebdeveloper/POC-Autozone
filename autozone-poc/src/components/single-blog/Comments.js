import React, { Fragment } from "react";
import "../single-blog/single-blog.css";
import Button from "../../utlis/button/Button";


const initialList = [];

const Comments = () => { 
  const [state, setState] = React.useState({
    name:'',
    email:'',
    comments:''
  });
  
  const [list, setList] = React.useState(initialList);

  const handleChange = (event) => {
    const values = event.target.value;
   };

  const handleSubmit = (event) => {
   
      setList(list.concat(state.name + " : " + state.comments));
      setState({name:'',email:'',comments:''})
    event.preventDefault();
  };
 
  if (!list.length) {
    return (
      <div>
        <div id="respond" className="comment-respond">
          <h3>Leave A Comment</h3>
        </div>
        <form className="comment-form" onSubmit={handleSubmit}>
          <div className="comment-input">
            <div className="form-group-name">
              <input
                type="text"
                placeholder="Name..."
                value={state.name}
                onChange={
                 (e)=> setState({nam:e.target.value})}
                className="form-control"
              />
            </div>
            <div className="form-group-email">
              <input
                type="text"
                placeholder="Email..."
                value={state.email}
                onChange={
                  (e)=>setState({email:e.target.value})}
                className="form-control"
              />
            </div>
          </div>
          <div className="form-group-textarea">
            <textarea
              value={state.comments}
              onChange={
                (e)=>setState({comments:e.target.value})}
              placeholder="Comment..."
              cols="98"
              rows="8"
            ></textarea>
          </div>

          
            <Button btnText="Submit Comment"/>

        </form>
      </div>
    );
  } else {
    return (
      <div>
        <h4 className="comment-count">{list.length} Comment</h4>
        <ul>
          {list.map((item) => (
            <div className="comments-list">
              <li key={item}>{item}</li>
            </div>
          ))}
        </ul>

        <div id="respond" className="comment-respond">
          <h3>Leave A Comment</h3>
        </div>
        <form className="comment-form" onSubmit={handleSubmit}>
          <div className="comment-input">
            <div className="form-group-name">
              <input
                type="text"
                placeholder="Name..."
                value={state.name}
                onChange={(e)=>
                  setState({name:e.target.value})}
                className="form-control"
              />
            </div>

            <div className="form-group-email">
              <input
                type="text"
                placeholder="Email..."
                value={state.email}
                onChange={
                  (e)=>setState({email:e.target.value})}
                className="form-control"
              />
            </div>
          </div>
          <div className="form-group-textarea">
            <textarea
              value={state.comments}
              onChange={
                (e)=>setState({comments:e.target.value})}
              placeholder="Comment..."
              cols="98"
              rows="8"
            ></textarea>
          </div>

         
            <Button btnText="Submit Comment"/>

        </form>
      </div>
    );
  }
};

export default Comments;
