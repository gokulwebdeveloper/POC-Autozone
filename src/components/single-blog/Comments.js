import React, { Fragment } from "react";
import "../single-blog/single-blog.css";
import Button from "../../utlis/button/Button";
import InputWithLabel from "../../utlis/formComponents/InputWithLabel";

const initialList = [];

const Comments = () => {
  const [state, setState] = React.useState({
    name: "",
    email: "",
    comments: "",
  });

  const [list, setList] = React.useState(initialList);

  const handleSubmit = (event) => {
   
    setList(list.concat(state.name + " : " + state.comments));
    setState({ name: "", email: "", comments: "" });
    event.preventDefault();
  };

  if (!list.length) {
    return (
      <div>
       <form className="comment-form" onSubmit={handleSubmit}>
        <h3>Leave A Comment</h3>
          <div className="col-25">
              <InputWithLabel
                inputId="Name"
                inputType="inputWithoutLabel"
                placeholder="Name..."
                onChange={(e) => setState({ ...state, name: e.target.value })}
                className="form-control"
              />
            </div>
            <div className="col-25">
              <InputWithLabel
                inputId="Email"
                inputType="inputWithoutLabel"
                placeholder="Email..."
                onChange={(e) => setState({ ...state, email: e.target.value })}
                className="form-control"
              />
            </div>
            
           <div className="col-75">
            <textarea className = "form-control" rows = "3"
             value={state.comments}
              onChange={(e) => setState({ ...state, comments: e.target.value })}
              placeholder="Comment..."
             
            ></textarea>
          </div>

          <Button btnText="Submit Comment" />
        </form>
      </div>
    );
  } else {
    return (
      <div>
        <h4 >{list.length} Comment</h4>
        <ul>
          {list.map((item) => (
            <div className="comments-list">
              <li key={item}>{item}</li>
            </div>
          ))}
        </ul>
        <form className="comment-form" onSubmit={handleSubmit}>
        <h3>Leave A Comment</h3>
           <div className="col-25">
               <InputWithLabel
                 inputId="Name"
                 inputType="inputWithoutLabel"
                 placeholder="Name..."
                 onChange={(e) => setState({ ...state, name: e.target.value })}
                 className="form-control"
               />
             </div>
             <div className="col-25">
               <InputWithLabel
                 inputId="Email"
                 inputType="inputWithoutLabel"
                 placeholder="Email..."
                 onChange={(e) => setState({ ...state, email: e.target.value })}
                 className="form-control"
               />
             </div>
            
            <div className="col-75">
             <textarea className = "form-control" rows = "3"
              value={state.comments}
               onChange={(e) => setState({ ...state, comments: e.target.value })}
               placeholder="Comment..."
              
             ></textarea>
           </div>

          <Button btnText="Submit Comment" />
        </form>
      </div>
    );
  }
};

export default Comments;
