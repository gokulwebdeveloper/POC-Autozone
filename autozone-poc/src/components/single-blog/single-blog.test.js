import React from "react";
import Singleblog from "./single-blog";
import { unmountComponentAtNode } from "react-dom";
import { shallow } from "enzyme";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { create } from "react-test-renderer";
import Comments from "./Comments";

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

describe("SingleBlog exists", () => {
  const wrapper = shallow(<Singleblog />);
  it("SingleBlog Exists", () => {
    expect(wrapper.exists()).toBeTruthy();
  });
  const wr = shallow(<Singleblog />);
  it("renders an image", () => {
    expect(wr.find("singleblogvideo")).toBeTruthy();
  });
  const sglblg = shallow(<Singleblog />);

  it("renders an image", () => {
    expect(sglblg.find("related-blog-post")).toBeTruthy();
  });

  test("SingleBlog renders a <Comments/>", () => {
    const sb = shallow(<Comments />);
    expect(sb.exists()).toBeTruthy;
  });
  const wrappersnap = shallow(<Singleblog />);
  it("SnapShotÂ Match", () => {
    expect(wrappersnap).toMatchSnapshot();
  });
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});
