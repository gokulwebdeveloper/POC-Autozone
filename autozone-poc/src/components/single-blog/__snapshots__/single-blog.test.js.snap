// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleBlog exists SnapShotÂ Match 1`] = `ShallowWrapper {}`;
